### PR TITLE
Update java_tools v13.9 / rules_java 7.12.0

### PR DESCRIPTION
**Pull request by iancha1992**
https://github.com/bazelbuild/java_tools/issues/92

### CODE REVIEW

**Comment by iancha1992** [comment]
I am trying to update the lockfile, but I get this error:
```
(base) heec-macbookpro:bazel heec$ bazel run //src/test/tools/bzlmod:update_default_lock_file
ERROR: For repository 'bazel_skylib', the root module requires module version bazel_skylib@1.6.1, but got bazel_skylib@1.7.1 in the resolved dependency graph.
ERROR: For repository 'io_bazel_skydoc', the root module requires module version stardoc@0.5.6, but got stardoc@0.7.0 in the resolved dependency graph.
ERROR: For repository 'rules_cc', the root module requires module version rules_cc@0.0.9, but got rules_cc@0.0.10 in the resolved dependency graph.
ERROR: For repository 'rules_proto', the root module requires module version rules_proto@6.0.0, but got rules_proto@6.0.2 in the resolved dependency graph.
ERROR: Error computing the main repository mapping: Direct dependency check failed.
```
Please advise @hvadehra Thanks!

**Comment by hvadehra** [comment]
Attempting to update them in https://github.com/bazelbuild/bazel/pull/23654, lets see how it goes.

**Comment by hvadehra** [comment]
@iancha1992 #23654 is merged, please try after rebasing this on HEAD.

**Comment by hvadehra** [comment]
@iancha1992 Should be fixed by https://github.com/bazelbuild/rules_java/commit/e4d1f09aeb143d20a21008793cb46d4b91c8005c. You'll need to update your rules_java branch (https://github.com/bazelbuild/rules_java/tree/java_v13.9) to HEAD.

**Comment by iancha1992** [comment]
@hvadehra I am experiencing another failure. Could you please take a look?
I've updated the rules_java branch by rebasing it like you suggested and tried to update the lockfile (please see below). Thanks!
```
(base) heec-macbookpro:bazel heec$ bazel run //src/test/tools/bzlmod:update_default_lock_file
ERROR: /private/var/tmp/_bazel_heec/1ec8945322d13a649f0866417f1bb8c5/external/rules_java+/java/common/proguard_spec_info.bzl:16:20: ProguardSpecProvider is experimental and thus unavailable with the current flags. It may be enabled by setting --experimental_google_legacy_api
ERROR: /Users/heec/Desktop/bazel/src/test/tools/bzlmod/BUILD:11:10: While resolving toolchains for target //src/test/tools/bzlmod:update_default_lock_file (bf2c9ef): invalid registered execution platform '//:default_host_platform': error loading package '': at /Users/heec/Desktop/bazel/tools/distributions/distribution_rules.bzl:15:6: at /private/var/tmp/_bazel_heec/1ec8945322d13a649f0866417f1bb8c5/external/rules_java+/java/defs.bzl:17:6: at /private/var/tmp/_bazel_heec/1ec8945322d13a649f0866417f1bb8c5/external/rules_java+/java/java_import.bzl:16:6: at /private/var/tmp/_bazel_heec/1ec8945322d13a649f0866417f1bb8c5/external/rules_java+/java/bazel/rules/bazel_java_import.bzl:22:6: at /private/var/tmp/_bazel_heec/1ec8945322d13a649f0866417f1bb8c5/external/rules_java+/java/common/rules/impl/bazel_java_import_impl.bzl:25:6: at /private/var/tmp/_bazel_heec/1ec8945322d13a649f0866417f1bb8c5/external/rules_java+/java/common/rules/impl/proguard_validation.bzl:20:6: compilation of module 'java/common/proguard_spec_info.bzl' failed
ERROR: Analysis of target '//src/test/tools/bzlmod:update_default_lock_file' failed; build aborted
INFO: Elapsed time: 0.201s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
ERROR: Build did NOT complete successfully
ERROR: Build failed. Not running target
```

File: MODULE.bazel [hunk 1]
```diff
@@ -45,6 +45,13 @@ single_version_override(
     version = "6.0",
 )
 
+archive_override(
+    module_name = "rules_java",
+    urls = ["https://github.com/bazelbuild/rules_java/archive/0355f81fb53a9ca267d69c0170d53a7b3f854d9d.tar.gz"],
```
Comment (hvadehra) [inline MODULE.bazel:50]: Looks like this hasn't been updated after you synced your branch?
I see your latest commit is:
https://github.com/bazelbuild/rules_java/commit/13e85b4ad6c142aea39605593e98c8d971981369

Comment (hvadehra) [reply MODULE.bazel:50]: @iancha1992 (same for workspace_deps.bzl below)

Comment (iancha1992) [reply MODULE.bazel]: Sorry, forgot about that! Thank you!

**Comment by iancha1992** [comment]
Duplicate: https://github.com/bazelbuild/bazel/pull/23738

**Pull request closed without merging**